newIDE/app/src/GameDashboard/GameThumbnail.js

```diff
@@ -1,54 +1,209 @@
 import * as React from 'react';
 import EmptyMessage from '../UI/EmptyMessage';
 import Paper from '../UI/Paper';
+import { useResponsiveWindowSize } from '../UI/Responsive/ResponsiveWindowMeasurer';
+import AuthenticatedUserContext from '../Profile/AuthenticatedUserContext';
+import { ColumnStackLayout } from '../UI/Layout';
+import RaisedButton from '../UI/RaisedButton';
+import useAlertDialog from '../UI/Alert/useAlertDialog';
+import {
+  createGameResourceSignedUrls,
+  updateGame,
+  type Game,
+} from '../Utils/GDevelopServices/Game';
+import { uploadBlobFile } from '../ExportAndShare/BrowserExporters/BrowserFileUploader';
+import { CorsAwareImage } from '../UI/CorsAwareImage';
 
 const styles = {
   image: {
     display: 'block',
-    objectFit: 'cover',
+    objectFit: 'scale-down', // Match gd.games format.
   },
   thumbnail: {
     // 16/9 format
     width: 272,
     height: 153,
   },
+  fullWidth: {
+    width: '100%',
+    height: 'auto',
+  },
 };
 
 type Props = {|
   thumbnailUrl?: string,
   gameName: string,
   background?: 'light' | 'medium' | 'dark',
+  project?: gdProject,
+  gameId: string,
+  canUpdateThumbnail?: boolean,
+  disabled?: boolean,
+  onGameUpdated?: (updatedGame: Game) => void,
+  onUpdatingGame?: (isUpdatingGame: boolean) => void,
 |};
 
 export const GameThumbnail = ({
+  project,
+  disabled,
+  canUpdateThumbnail,
   thumbnailUrl,
   gameName,
+  gameId,
+  onGameUpdated,
+  onUpdatingGame,
   background = 'light',
-}: Props) =>
-  thumbnailUrl ? (
-    <img
-      src={thumbnailUrl}
-      style={{
-        ...styles.image,
-        ...styles.thumbnail,
-      }}
-      alt={gameName}
-      title={gameName}
-    />
-  ) : (
-    <Paper
-      variant="outlined"
-      style={{
-        ...styles.thumbnail,
-        whiteSpace: 'normal',
-        display: 'flex',
-      }}
-      background={background}
-    >
-      <EmptyMessage>
-        <Trans>No thumbnail set</Trans>
-      </EmptyMessage>
-    </Paper>
+}: Props) => {
+  const { isMobile } = useResponsiveWindowSize();
+  const { profile, getAuthorizationHeader } = React.useContext(
+    AuthenticatedUserContext
+  );
+  const [isLoading, setIsLoading] = React.useState(false);
+  const { showAlert } = useAlertDialog();
+  const gamesPlatformThumbnailFileInputRef = React.useRef<HTMLInputElement | null>(
+    null
+  );
+
+  const updateGameThumbnail = async () => {
+    if (!profile) {
+      return;
+    }
+
+    if (
+      !gamesPlatformThumbnailFileInputRef.current ||
+      !gamesPlatformThumbnailFileInputRef.current.files ||
+      !gamesPlatformThumbnailFileInputRef.current.files[0]
+    ) {
+      console.error("Could't find selected file. Aborting thumbnail update.");
+      return;
+    }
+    const file = gamesPlatformThumbnailFileInputRef.current.files[0];
+    const fileType = file.type;
+    if (
+      !fileType ||
+      (fileType !== 'image/png' &&
+        fileType !== 'image/jpeg' &&
+        fileType !== 'image/webp')
```

Comment:
Should we prevent webp format? It's not supported by every browser

Reply:
I think that's ok, it's here in Safari 16 (and Safari 15 on certain macos version), so overall it should be ok
<img width="1345" alt="image" src="https://github.com/user-attachments/assets/8b45e236-3b70-4b0c-9f33-a81291a79ba0">